runtime/src/lib.rs

```diff
@@ -306,6 +308,56 @@ pub trait Clock: Clone + Send + Sync + 'static {
     }
 }
 
+cfg_if::cfg_if! {
+    if #[cfg(feature = "external")] {
+        /// Interface that runtimes can implement to constrain the execution latency of a future.
+        pub trait Pacer: Clock + Clone + Send + Sync + 'static {
+            /// Defer completion of a future until a randomly selected delay within `range` has elapsed. If
```

Comment:
This bit is confusing to me as a user: what's the reason for selecting a random value within `range`? Why is that desirable?

If I read the rest correctly:

1. the runtime attempts to poll the first time after `delay` (specifically, a value that falls into `delay` but is chosen at random),
2. and if the wrapped future isn't ready, it polls until it returns `Poll::Ready`?

It sounds not ideal if the runtime blocks until fut is ready.